src/selectors/getChildren.ts

```diff
@@ -74,3 +80,38 @@ export const getChildrenRanked = (state: State, context: Context): Child[] =>
 /** Returns the first visible child of a context. */
 export const firstVisibleChild = (state: State, context: Context) =>
   getChildrenSorted(state, context)[0]
+
+/** Checks if a child lies within the cursor path. */
+const isChildInCursor = (state: State, path: Path, simplePath: SimplePath, child: Child | ThoughtContext) => {
+  const showContexts = isContextViewActive(state, pathToContext(path))
+  const childSimplePath = getChildPath(state, child, simplePath, showContexts)
+  const childPath = appendChildPath(state, childSimplePath, path)
+  return state.cursor && equalThoughtRanked(state.cursor[childPath.length - 1], head(childPath))
+}
+
+/** Returns head of context if parent has active context view. */
+const getResolvedValue = (state: State, path: Path, child: Child | ThoughtContext) => {
+  const showContexts = isContextViewActive(state, pathToContext(path))
+  return showContexts
+    ? head((child as ThoughtContext).context)
+    : (child as Child).value
+}
+
+/** Checks if the child is visible and also checks if the child lies within the cursor. */
+const isChildVisibleWithCursorCheck = _.curry((state: State, path: Path, simplePath: SimplePath, child: Child | ThoughtContext) => {
+  return state.showHiddenThoughts ||
+  isChildVisible(state, pathToContext(simplePath), { value: getResolvedValue(state, path, child), rank: child.rank }) ||
+  isChildInCursor(state, path, simplePath, child)
+})
+
+/**
+ * Returns only visible thoughts from the children array. It checks if a thought lies within the cursor path.
+ *
+ * @param state - Redux state.
+ * @param path - Parent's path.
+ * @param simplePath - Parent's simple path.
+ * @param children - Children of the given simplePath's context.
+ */
+export const getChildrenWithCursorCheck = (state: State, path: Path, simplePath: SimplePath, children: (Child | ThoughtContext)[]) => {
+  return children.filter(isChildVisibleWithCursorCheck(state, path, simplePath))
+}
```

Comment:
If it has to be passed `children`, it's not really `getting` the children, so this isn't the best name. This function is just wrapping filter and hiding `isChildVisibleWithCursorCheck`, so I don't think it adds much value. It may be unnecessary. Unless it is able to query the children internally, we should just export `isChildVisibleWithCursorCheck` and use that.

Reply:
Agreed! I think we should just export `isChildVisibleWithCursorCheck` and use it where necessary.

Reply:
Cool, I'll do this when I merge `dev`.